### PR TITLE
Parse liberty files

### PR DESCRIPTION
**Pull request by jameswolfeoliver**
Parse liberty files. Groups are parsed into a Group object via it's namespace dict which is janky but idgaf. End result means we can use nice `.` notation such as `cell.name` or `cell.direction`. Checkout the test file for more implementation details.

**Basic language (approximately)**
```
liberty_object -> statement + liberty_object
liberty_object -> EOF
statement -> group_statement
statement -> simple_attribute
statement -> complex_attribute
group_statement -> group_name + ( + name + ) + { + list[statement] + }
group_name -> string
name -> string
simple_attribute -> attribute_name + : + attribute_value
attribute_name -> string
attribute_value -> string
complex_attribute -> attribute_name + ( + parameters + ) + ;
attribute_name -> string
parameters -> parameter + optional(, + parameters)
parameter -> [ + parameters + ]
```

### CODE REVIEW

**Review by thrile — DISMISSED** [review]
I added my comments to Slack

File: python/Pipfile [hunk 1]
```diff
@@ -10,6 +10,7 @@ mypy = "*"
 [packages]
 numpy = "*"
```
Comment (thrile) [inline python/Pipfile:11]: Are we actually using this yet?

File: python/Pipfile [hunk 2]
```diff
@@ -10,6 +10,7 @@ mypy = "*"
...
 pylint = "*"
```
Comment (thrile) [inline python/Pipfile:12]: Is this actually needed in the release version or could it be moved to dev?

**Review by thrile — APPROVED** [review]

File: python/Pipfile (hunk 1 above)
Comment (jameswolfeoliver) [reply python/Pipfile:11]: No not yet

File: python/Pipfile (hunk 2 above)
Comment (jameswolfeoliver) [reply python/Pipfile:12]: probs could be moved to dev

**Pull request merged**